### PR TITLE
[Media] Add react/dynamic filter form

### PR DESCRIPTION
**Pull request by alisterdev**
- [x] Rebuild main page using new generic React components (i.e `Tabs`, `FilterForm`, etc)
- [x] Add additional fields to filter form (Redmine 11753)
- [x] Remove unused tpls and CSS

### CODE REVIEW

**Comment by ridz1208** [comment]
- 2 empty fields and redundant filetypes
screenshot from 2017-02-20 22-11-17

**Comment by ridz1208** [comment]
@alisterdev other than stuff above, **extremely cool**

**Comment by alisterdev** [comment]
@ridz1208 
1. Redundant file types comme from your data, you might have some old records with manually entered types. We should correct that, and maybe change `file_type` to enum of accepted formats

2. Good catch. A helper function isn't decoding strings with spaces in them "CCNA%20Testing%20Site". Will fix

**Comment by alisterdev** [comment]
@ridz1208 I fixed the bug you mentioned and got Tabs PR (the one this PR relied on) merged, so this it can be thoroughly retested now

**Comment by ridz1208** [comment]
@alisterdev file types should be a separate table in the database with a foreign key from the field in media (i think that table already exists in loris for some other uploaders)

File: modules/media/templates/menu_media.tpl [hunk 1]
```diff
@@ -1,29 +1,4 @@
...
-  <!-- Upload Media -->
-  <div role="tabpanel" class="tab-pane" id="upload">
-      {include file='../part/file_upload.tpl'}
-  </div>
-
-</div>
+{*
+  Everything is rendered in mediaIndex.js now!
+  I'm here because currently Loris breaks without me :(
+*}
```
Comment (jstirling91) [inline modules/media/templates/menu_media.tpl]: Can you try overwriting the display function to return an empty string in the NDB_Menu_Filter_media.class.inc and remove this file to see if that doesn't break Loris?

Comment (alisterdev) [inline modules/media/templates/menu_media.tpl]: @jstirling91 Great tip, it does work indeed, so I was able to completely get rid of smarty templates in media

**Comment by alisterdev** [comment]
@ridz1208 & @jacobpenny Please re-review me

**Comment by ridz1208** [comment]
@alisterdev 

 - in dropdowns, when "any" is selected, the field stays empty (might be desired behaviour) 
 - more issues with back button
  - before
  - after
 - no 'clear form' button ? you have to manually undo all fields entered
 - is the upload form centered on purpose ? (wannabe google)

**Comment by alisterdev** [comment]
@ridz1208 

1. Removed all uses of "any" since it was just duplicating behaviour of selecting an empty option
2. Problems caused by `&` in the value string `#fixed `
3. Clear button added
4. Was experimenting with new style, but brought the form back to the left now `#theycanthandlemyswag`
5. It needs to be in this format, since `StaticDataTable` converts all strings to camelCase and it does not work nicely with all caps abbreviations.

File: modules/media/jsx/uploadForm.js [hunk 2]
```diff
@@ -80,92 +80,93 @@ class MediaUploadForm extends React.Component {
...
+            <StaticElement
+              label="Note"
+              text={helpText}
+            />
+            <SelectElement
+              name="pscid"
+              label="PSCID"
+              options={this.state.Data.candidates}
+              onUserInput={this.setFormData}
+              ref="pscid"
```
Comment (jacobpenny) [inline modules/media/jsx/uploadForm.js]: Do we still need these refs?

File: modules/media/php/NDB_Menu_Filter_media.class.inc [hunk 3]
```diff
@@ -82,20 +100,29 @@ class NDB_Menu_Filter_Media extends NDB_Menu_Filter
...
-        $this->addSelect('instrument', 'Instrument:', $instrumentList);
-        $this->addSelect('visit_label', 'Visit Label:', $visitList);
-        $this->addSelect('for_site', 'For Site:', $siteList);
+        $this->addBasicText('pSCID', 'PSCID', ["size" => 9, "maxlength" => 7]);
+        $this->addBasicText('fileName', 'File Name');
+        $this->addSelect('fileType', 'File type', $fileTypeList);
+        $this->addSelect('visitLabel', 'Visit Label', $visitList);
+        $this->addSelect('site', 'For Site', $siteList);
+        $this->addBasicText('uploadedBy', 'Uploaded By');
+        $this->addSelect('instrument', 'Instrument', $instrumentList);
```
Comment (jacobpenny) [inline modules/media/php/NDB_Menu_Filter_media.class.inc]: This is all rendered via React isn't it? Do we need to keep these calls to LorisForm around?
Is it a round-about way of providing the data needed for the AJAX request?

Comment (alisterdev) [inline modules/media/php/NDB_Menu_Filter_media.class.inc]: I've been asked to allow for integration of a react filter with a minimum of modification on the end-user/programmer part. So this is actually testing the case where programmer won't have to update the php code to get react filters working.

File: modules/media/jsx/uploadForm.js (hunk 2 above)
Comment (alisterdev) [inline modules/media/jsx/uploadForm.js]: Yep, they are still used across `uploadForm`. We could potentially get rid of them in the future.

File: modules/media/php/NDB_Menu_Filter_media.class.inc (hunk 3 above)
Comment (jacobpenny) [inline modules/media/php/NDB_Menu_Filter_media.class.inc]: Ah gotcha

**Comment by alisterdev** [comment]
@ridz1208 Passed manual tests?

File: modules/media/php/NDB_Menu_Filter_media.class.inc [hunk 4]
```diff
@@ -48,6 +49,17 @@ class NDB_Menu_Filter_Media extends NDB_Menu_Filter
...
     /**
+     * Override default behaviour, since the page is loaded from mediaIndex.js
+     *
+     * @return void
+     * @access public
+     */
+    function display()
+    {
+        return null;
+    }
```
Comment (driusan) [inline modules/media/php/NDB_Menu_Filter_media.class.inc]: I don't really like this

Comment (alisterdev) [inline modules/media/php/NDB_Menu_Filter_media.class.inc]: I don't see why overriding default behaviour through polymorphism worries you

**Pull request merged**